controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -83,6 +123,46 @@ bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::st
     return message_com::send_cmdu(sd, cmdu_tx);
 }
 
+bool wfa_ca::parse_params(std::unordered_map<std::string, std::string> &params,
```

Comment:
It is very weird to mix the introduction of this function in a commit that also does 4 other things, but does not actually call this function.

Please please please do 1 thing per commit. I spend too much time reviewing because of this. When you have just implemented it, you understand what the code is doing, so it is easy to see which parts belong together and put them together in one commit. When reviewing, you have no idea what is going on, so trying to see if what has changed is indeed the right thing to do takes a lot more mental energy and thus time.

Remember, the point of reviewing is to check if the change what was done is *the right thing* and *for the right reasons*. The first part is easier to validate when the commit is properly split up. The second part is easier to validate when there is a commit message that explains the reasons. If the reviewer has to make a guess at all of this, it will take a lot more time.

Reply:
Yeah, I try to make infrastructure changes that will be used in the follow-up commit -preparation commit.
I guess I understand why it is more default to review change that it is not used.
In CONTRIBUTING.md it is said:

> "Since commits should be split up, there will be many commits that are not useful on their own, but just prepare for another commit. In that case, the preparatory commits should refer to what they prepare for."

I see now that I forgot to mention in the commit message that it is a preparatory commit.
Next time :)